### PR TITLE
crowbar: Workaround to avoid the drift caused by sync marks (bsc#935462)

### PR DESCRIPTION
**Pull request by vuntz**
Because each chef-client runs executes everything in all cookbooks, and because
the crm calls slow things down considerably, we have a drift issue where the
founder (which does much more work since it's the only one doing the crm calls)
goes much slower than the other nodes. This can impact things badly when it's
so slow that it triggers the timeout in the sync marks.

The workaround here is to reset the sync marks on the founder, so that all
nodes wait on the sync marks and the drift is only happening between sync
marks.

The goal is to only do this when applying a proposal, so that other chef-client
runs are not blocked waiting for sync marks.

Unfortunately, there are cases where we cannot do that: for swift, it's
likely that swift-ring-compute would be on a cluster node, and since
that role is run first, the cluster node would be alone in running chef
and blocking forever. This means that this small helper here will have
to be called explicitly by all barclamps for which there's no such issue
-- afaict, swift is the only one with an issue.

All of this should be removed when we get proper crowbar orchestration.

https://bugzilla.suse.com/show_bug.cgi?id=935462

### CODE REVIEW

File: crowbar_framework/app/models/pacemaker_service_object.rb [hunk 1]
```diff
@@ -133,6 +133,37 @@ def role_expand_elements(role, role_name)
...
+  # where the founder (which does much more work since it's the only one doing
+  # the crm calls) goes much slower than the other nodes. This can impact
+  # things badly when it's so slow that it triggers the timeout in the sync
+  # marks.
+  # The workaround here is to reset the sync marks on the founder, so that all
+  # nodes wait on the sync marks and the drift is only happening between sync
+  # marks.
+  # The goal is to only do this when applying a proposal, so that other
+  # chef-client runs are not blocked waiting for sync marks.
+  def reset_sync_marks_on_clusters_founders(elements)
```
Comment (aplanas) [inline crowbar_framework/app/models/pacemaker_service_object.rb]: Ah here it is

**Comment by jsuchome** [comment]
+1 (not tested, but the code looks correct)

**Comment by aplanas** [comment]
+1

**Comment by aplanas** [comment]
+1

**Comment by aplanas** [comment]
Gating fails with: `Error: Committing the crowbar 'default' proposal for 'dns' failed`, too early. But looks unrelated

**Comment by vuntz** [comment]
Merging as gating now deploys (up to nova as this point in time), and this is just a new method that can't cause failures in the remaining part of the gating.

**Comment by aspiers** [comment]
Retrospective -1 purely due to the newly introduced long lines :-p  (otherwise +1)

**Comment by vuntz** [comment]
> due to the newly introduced long lines :-p

I love long lines ;-) We can fix that up now if you want.

**Comment by aspiers** [comment]
Because they annoy the English guy, no doubt ;-p  Yes please.

**Comment by vuntz** [comment]
> Because they annoy the English guy, no doubt ;-p Yes please.

Actually, that's fixed in https://github.com/crowbar/crowbar-ha/pull/61

**Pull request merged**